### PR TITLE
Crossdomxaction

### PR DESCRIPTION
**Pull request by bhavaniravichandran**
tests cross domain transaction with domain in image with 2 domains running in the same cluster different namespaces.

### CODE REVIEW

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java [hunk 1]
```diff
@@ -0,0 +1,400 @@
...
+    createDomain(domainUid2, domain2Namespace, domain2AdminSecretName, domain2Image);
+
+    logger.info("Getting admin server external service node port");
+    int adminServiceNodePort = assertDoesNotThrow(
+        () -> getServiceNodePort(domain1Namespace, domain1AdminServerPodName + "-external", "default"),
+        "Getting admin server node port failed");
+
+    String curlRequest = String.format("curl -v --show-error --noproxy '*' "
+            + "http://%s:%s/TxForward/TxForward?urls=t3://%s.%s:7001,t3://%s1.%s:8001,t3://%s1.%s:8001",
+             K8S_NODEPORT_HOST, adminServiceNodePort, domain1AdminServerPodName, domain1Namespace,
```
Comment (ajsomogyi) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java:242]: add domain2-managed-server2 to end of URL list

File: new-integration-tests/src/test/resources/apps/txpropagate/build.xml [hunk 2]
```diff
@@ -0,0 +1,102 @@
+<!-- Copyright (c) 2020, Oracle Corporation and/or its affiliates.
+     Licensed under the Universal Permissive License v 1.0 as shown at
+          https://oss.oracle.com/licenses/upl.-->
+
+<project name="txpropagate.app" default="build">
```
Comment (ajsomogyi) [inline new-integration-tests/src/test/resources/apps/txpropagate/build.xml:5]: Remove txpropagate application

File: new-integration-tests/src/test/resources/apps/txforward/remotesync/src/example/RemoteSync.java [hunk 3]
```diff
@@ -0,0 +1,10 @@
+package example;
+
+import java.rmi.Remote;
+import java.rmi.RemoteException;
+
+public interface RemoteSync extends Remote {
+  public static final String JNDINAME = "propagate.RemoteSync";
```
Comment (ajsomogyi) [inline new-integration-tests/src/test/resources/apps/txforward/remotesync/src/example/RemoteSync.java:7]: change JNDIName to txforward.RemoteSync

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java (hunk 1 above)
Comment (bhavaniravichandran) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java:240]: added domain2-managed-server2 to the url list

File: new-integration-tests/src/test/resources/apps/txpropagate/build.xml (hunk 2 above)
Comment (bhavaniravichandran) [reply new-integration-tests/src/test/resources/apps/txpropagate/build.xml]: removed txpropagate app

File: new-integration-tests/src/test/resources/apps/txforward/remotesync/src/example/RemoteSync.java (hunk 3 above)
Comment (bhavaniravichandran) [reply new-integration-tests/src/test/resources/apps/txforward/remotesync/src/example/RemoteSync.java]: fixed

**Comment by rjeberhard** [comment]
@bhavaniravichandran, I'm not sure how to resolve the conflict in this change.

**Comment by bhavaniravichandran** [comment]
> @bhavaniravichandran, I'm not sure how to resolve the conflict in this change.

@rjeberhard It wasn't ready to be merged. I am dependent on sankar's change you just merged. How should I proceed now?

**Comment by rjeberhard** [comment]
@bhavaniravichandran, on your branch `git merge origin/develop`, resolve the conflicts, commit and push. This will automatically update the PR.

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java [hunk 4]
```diff
@@ -0,0 +1,402 @@
...
+
+    logger.info("Creating unique namespace for Domain");
+    assertNotNull(namespaces.get(1), "Namespace list is null");
+    domain1Namespace = namespaces.get(1);
+
+    logger.info("Creating unique namespace for Domain");
+    assertNotNull(namespaces.get(2), "Namespace list is null");
+    domain2Namespace = namespaces.get(2);
+
+    // Now that we got the namespaces for both the domains,w e need to update the model properties
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java:128]:
```suggestion
    // Now that we got the namespaces for both the domains, we need to update the model properties
```

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java [hunk 5]
```diff
@@ -0,0 +1,402 @@
...
+    // file with the namespaces. for cross domain transaction to work, we need to have the externalDNSName
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java:129]:
```suggestion
    // file with the namespaces. For cross domain transaction to work, we need to have the externalDNSName
```

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java [hunk 6]
```diff
@@ -0,0 +1,386 @@
...
+
+    // create admin credential secret for domain2
+    logger.info("Create admin credential secret for domain2");
+    String domain2AdminSecretName = domainUid2 + "-weblogic-credentials";
+    assertDoesNotThrow(() -> createSecretWithUsernamePassword(
+        domain2AdminSecretName, domain2Namespace, ADMIN_USERNAME_DEFAULT, ADMIN_PASSWORD_DEFAULT),
+        String.format("createSecret %s failed for %s", domain2AdminSecretName, domainUid2));
+
+    //createImageVerify expects the location of the ear file
+    //String appSource = PV_ROOT + "/applications/" + WDT_APP_NAME + "/" + WDT_APP_NAME + ".ear";
```
Comment (sankarpn) [inline new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java:214]: remove commented code

File: new-integration-tests/src/test/resources/apps/txforward/TxForward/WebContent/META-INF/application.xml [hunk 7]
```diff
@@ -0,0 +1,10 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (sankarpn) [inline new-integration-tests/src/test/resources/apps/txforward/TxForward/WebContent/META-INF/application.xml:1]: missing copyright

File: new-integration-tests/src/test/resources/apps/txforward/TxForward/WebContent/META-INF/weblogic-application.xml [hunk 8]
```diff
@@ -0,0 +1,7 @@
+<?xml version = '1.0' ?>
```
Comment (sankarpn) [inline new-integration-tests/src/test/resources/apps/txforward/TxForward/WebContent/META-INF/weblogic-application.xml:1]: missing copyright

File: new-integration-tests/src/test/resources/apps/txforward/TxForward/src/example/TxForward.java [hunk 9]
```diff
@@ -0,0 +1,71 @@
+package example;
```
Comment (sankarpn) [inline new-integration-tests/src/test/resources/apps/txforward/TxForward/src/example/TxForward.java:1]: missing copyright

File: new-integration-tests/src/test/resources/apps/txforward/remotesync/src/example/LifecycleListenerImpl.java [hunk 10]
```diff
@@ -0,0 +1,18 @@
+package example;
```
Comment (sankarpn) [inline new-integration-tests/src/test/resources/apps/txforward/remotesync/src/example/LifecycleListenerImpl.java:1]: missing copyright

File: new-integration-tests/src/test/resources/apps/txforward/remotesync/src/example/RemoteSync.java [hunk 11]
```diff
@@ -0,0 +1,10 @@
+package example;
```
Comment (sankarpn) [inline new-integration-tests/src/test/resources/apps/txforward/remotesync/src/example/RemoteSync.java:1]: missing copyright

File: new-integration-tests/src/test/resources/apps/txforward/remotesync/src/example/RemoteSyncImpl.java [hunk 12]
```diff
@@ -0,0 +1,106 @@
+package example;
```
Comment (sankarpn) [inline new-integration-tests/src/test/resources/apps/txforward/remotesync/src/example/RemoteSyncImpl.java:1]: missing copyright

File: new-integration-tests/src/test/resources/apps/txforward/remotesync/src/example/Utils.java [hunk 13]
```diff
@@ -0,0 +1,29 @@
+package example;
```
Comment (sankarpn) [inline new-integration-tests/src/test/resources/apps/txforward/remotesync/src/example/Utils.java:1]: missing copyright

Comment (bhavaniravichandran) [reply new-integration-tests/src/test/resources/apps/txforward/remotesync/src/example/Utils.java:4]: done

File: new-integration-tests/src/test/resources/apps/txforward/remotesync/src/example/RemoteSyncImpl.java (hunk 12 above)
Comment (bhavaniravichandran) [reply new-integration-tests/src/test/resources/apps/txforward/remotesync/src/example/RemoteSyncImpl.java:4]: done

File: new-integration-tests/src/test/resources/apps/txforward/remotesync/src/example/RemoteSync.java (hunk 11 above)
Comment (bhavaniravichandran) [reply new-integration-tests/src/test/resources/apps/txforward/remotesync/src/example/RemoteSync.java:4]: added copyright

File: new-integration-tests/src/test/resources/apps/txforward/remotesync/src/example/LifecycleListenerImpl.java (hunk 10 above)
Comment (bhavaniravichandran) [reply new-integration-tests/src/test/resources/apps/txforward/remotesync/src/example/LifecycleListenerImpl.java:4]: added copyright

File: new-integration-tests/src/test/resources/apps/txforward/TxForward/src/example/TxForward.java (hunk 9 above)
Comment (bhavaniravichandran) [reply new-integration-tests/src/test/resources/apps/txforward/TxForward/src/example/TxForward.java:4]: added copyright

File: new-integration-tests/src/test/resources/apps/txforward/TxForward/WebContent/META-INF/weblogic-application.xml (hunk 8 above)
Comment (bhavaniravichandran) [reply new-integration-tests/src/test/resources/apps/txforward/TxForward/WebContent/META-INF/weblogic-application.xml:1]: added copyright

File: new-integration-tests/src/test/resources/apps/txforward/TxForward/WebContent/META-INF/application.xml (hunk 7 above)
Comment (bhavaniravichandran) [reply new-integration-tests/src/test/resources/apps/txforward/TxForward/WebContent/META-INF/application.xml:1]: added copyright

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java (hunk 6 above)
Comment (bhavaniravichandran) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java]: removed the 2 lines

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java (hunk 5 above)
Comment (bhavaniravichandran) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java]: fixed

File: new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java (hunk 4 above)
Comment (bhavaniravichandran) [reply new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItCrossDomainTransaction.java]: fixed

**Pull request merged**